### PR TITLE
Added new documentation for back end menu item (Contao 4.9 or higher)

### PR DESCRIPTION
**Pull request by fabcocco**
Creating Backend Menu Item in Contao 4.9

### CODE REVIEW

**Comment by fritzmg** [comment]
Thank you for your contribution :). Your pull request contains formatting changes as well as new content. It would be better make two pull requests: one for improving the formatting and one for the additional new content. Also your Pull Request title should be changed to something more useful ;)

**Comment by theKalash** [comment]
We removed the "improved formatting" for now ;)

File: docs/dev/guides/backend-routes.md [hunk 1]
```diff
@@ -111,6 +111,10 @@ Most of the time you probably want to add a menu entry for your back end module.
 Since the back end menu can be extended with an `EventListener` we can easily
 create one that listens for the menu event to be dispatched.
 
+{{% notice warning %}}
+The following functions works in Contao **>4.9** for the Version **4.9** use the BackendMenuListener.php from above.
```
Comment (fritzmg) [inline docs/dev/guides/backend-routes.md]: Don't you mean it only works in Contao 4.9 and previous versions need to use the example above?

**Review by fritzmg — CHANGES_REQUESTED** [review]

File: docs/dev/guides/backend-routes.md (hunk 1 above)
Comment (theKalash) [reply docs/dev/guides/backend-routes.md]: Yes!

Comment (theKalash) [reply docs/dev/guides/backend-routes.md]: and above should probably be "below"  referring to the new block we added

Comment (theKalash) [reply docs/dev/guides/backend-routes.md]: The original example only works for Versions up to 4.8.* (before the menu refactoring), the example we added is for 4.9 and above.

**Comment by fritzmg** [comment]
@leofeyer but isn't it a BC break, that the previous example does not work any more?

**Comment by theKalash** [comment]
So from my knowledge, this is was changed between 4.8 and 4.9:

With 4.8 the call to `$event->getTree();` returned the sidebar menu. 
With 4.9 it returns the top-bar menu, and thus `$contentNode = $tree->getChild('content');` returns null and calling `->addChild()` fails. 

We also noticed that now that we add our own content Node, it appears in both, the top- and the sidebar.

**Comment by leofeyer** [comment]
Here is how you retrieve the sidebar menu in Contao 4.9:

```php
    public function addCustomItem(MenuEvent $event): void
    {
        $tree = $event->getTree();

        if ('mainMenu' !== $tree->getName()) {
            return;
        }

        $contentNode = $tree->getChild('content');
        // ...
```

The existing example only works accidentally, because in Contao 4.8 there is only one menu. 🤷‍♂

**Comment by fritzmg** [comment]
So is there a way that works in Contao 4.4, 4.8 and 4.9? If yes, then only that way should get documented.

**Comment by leofeyer** [comment]
Well, the menu name in Contao 4.8 was `root` instead of `mainMenu`, so there is no identical solution for both versions. However, Contao 4.8 will reach its EOL in 4 weeks, so we should only document how it works in Contao 4.9.

**Comment by theKalash** [comment]
Well inserting 

```
        if ('mainMenu' !== $tree->getName()) {
            return;
        }
```
Should be all that is necessary to update the existing example for 4.9. So I think we can disregard this PR

Though, using the setters on the factory seems much nicer than the array structure that was used before.

**Comment by fritzmg** [comment]
> Well, the menu name in Contao 4.8 was `root` instead of `mainMenu`, so there is no identical solution for both versions. However, Contao 4.8 will reach its EOL in 4 weeks, so we should only document how it works in Contao 4.9.

I see, so this was not possible in Contao 4.4 at all? Yeah, then we should simply change the current example to how it would work in 4.9.

**Comment by theKalash** [comment]
We updated it to only include the condition for the mainMenu and the new method based syntax. That should fix it for 4.9 and make it slightly prettier.

**Review by fritzmg — APPROVED** [review]

**Pull request merged**